server/py/services/api/main.py

```diff
@@ -554,7 +554,13 @@ def _start_periodic_partition_management(self):
                 f"Starting periodic partition management for table {table_name}",
                 retention_days=retention_days,
             )
-            interval_in_seconds = retention_days * 24 * 60 * 60
+            partition_interval = framework.db.session.run_function_with_new_db_session(
+                services.api.utils.db.partitioner.MySQLPartitioner().get_partition_interval,
+                table_name=table_name,
+            )
+            interval_in_seconds = int(
+                partition_interval.as_duration().total_seconds() / 2
```

Comment:
In the case where partition_interval is not DAY, MONTH, or YEARWEEK, it will return None, and attempting to call total_seconds() on None will result in an error. It may be better to raise a ValueError with the message 'Unsupported partition interval' in the as_duration function in that case.

Reply:
It's only half true - 
If we have a valid partition interval then it is an `StrEnum` so it must be one of the valid options.
However - `get_partition_interval` can possibly return `None`, and then we will get an error:
```
AttributeError: 'NoneType' object has no attribute 'as_duration'
```
So I agree there should be some defense here.

Reply:
Changed the code of `from_function` to make it always either return `PartitionInterval` or raise an exception